HiPMobile/HiPMobile.Droid/Activities/RouteNavigationActivity.cs

```diff
@@ -335,51 +332,54 @@ private void UpdateInstructions (GeoPoint currentLocation, RoadNode currentNode)
             }*/
         }
 
-        #region notImplented
+        protected override void OnDestroy()
+        {
+            GpsTracker.Unregister();
+
+            base.OnDestroy();
+        }
 
-        public void OnProviderDisabled (string provider)
+        public void OnProviderDisabled(string provider)
         {
-            Toast.MakeText (this, "GPS Disabled",
-                            ToastLength.Short).Show ();
+            Toast.MakeText(this, "GPS Disabled",
+                            ToastLength.Short).Show();
         }
 
-        public void OnProviderEnabled (string provider)
+        public void OnProviderEnabled(string provider)
         {
-            Toast.MakeText (this, "GPS Enabled",
-                            ToastLength.Short).Show ();
+            Toast.MakeText(this, "GPS Enabled",
+                            ToastLength.Short).Show();
         }
 
-        public void OnStatusChanged (string provider, [GeneratedEnum] Availability status, Bundle extras)
+        public void OnStatusChanged(string provider, [GeneratedEnum] Availability status, Bundle extras)
```

Comment:
Is it necessary that we give the user this information? Especially something like "Status Changed: Out of Service" is very generic so that the user will be confused. Maybe we should give information like "Location not available" or something similar